go/fixchain/main/fixchain.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/google/certificate-transparency/go/fixchain"
+	"github.com/google/certificate-transparency/go/fixchain/ratelimiter"
+	"github.com/google/certificate-transparency/go/x509"
+)
+
+// Assumes chains to be stores in a file in JSON encoded with the certificates
+// in DER format.
+func processChains(file string, fl *fixchain.FixAndLog) {
+	f, err := os.Open(file)
+	if err != nil {
+		log.Fatalf("Can't open %s: %s", f, err)
+	}
+	defer f.Close()
+
+	type Chain struct {
+		Chain [][]byte
+	}
+
+	dec := json.NewDecoder(f)
+	for {
+		var m Chain
+		if err := dec.Decode(&m); err == io.EOF {
+			break
+		} else if err != nil {
+			log.Fatal(err)
+		}
+		var chain []*x509.Certificate
+		for _, derBytes := range m.Chain {
+			cert, err := x509.ParseCertificate(derBytes)
+			switch err.(type) {
+			case nil, x509.NonFatalErrors:
+				// ignore
+			default:
+				log.Fatalf("can't parse certificate: %s %#v",
+					err, derBytes)
+			}
+
+			chain = append(chain, cert)
+		}
+		fl.QueueAllCertsInChain(chain)
+	}
+}
+
+func contentStore(baseDir string, subDir string, content []byte) {
```

Comment:
This is a function to save the errors that are spat out by the FixAndLog to a directory.  Note:  this is the simple open source main package.  People using this would most likely alter contentStore as they wish to store the errors wherever/however they need to.  Both logStringErrors() and logJSONErrors() use this function as a way of storing the FixErrors.